src/measure_theory/lp_space.lean

```diff
@@ -92,11 +92,80 @@ lemma zero_mem_ℒp (hp0_lt : 0 < p) : mem_ℒp (0 : α → E) p μ :=
 @[simp] lemma snorm_zero (hp0_lt : 0 < p) : snorm (0 : α → F) p μ = 0 :=
 by simp [snorm, hp0_lt]
 
+/-- When `μ = 0`, we have `∫ f^p ∂μ = 0`. `snorm f p μ` is then `0`, `1` or `⊤` depending on `p`. -/
+lemma snorm_measure_zero_of_pos {f : α → F} (hp_pos : 0 < p) : snorm f p 0 = 0 :=
+by simp [snorm, hp_pos]
+
+/-- When `μ = 0`, we have `∫ f^p ∂μ = 0`. `snorm f p μ` is then `0`, `1` or `⊤` depending on `p`. -/
+lemma snorm_measure_zero_of_exponent_zero {f : α → F} : snorm f 0 0 = 1 := by simp [snorm]
+
+/-- When `μ = 0`, we have `∫ f^p ∂μ = 0`. `snorm f p μ` is then `0`, `1` or `⊤` depending on `p`. -/
+lemma snorm_measure_zero_of_neg {f : α → F} (hp_neg : p < 0) : snorm f p 0 = ⊤ :=
+by simp [snorm, hp_neg]
+
 end zero
 
 @[simp] lemma snorm_neg {f : α → F} : snorm (-f) p μ = snorm f p μ :=
 by simp [snorm]
 
+section opens_measurable_space
+variable [opens_measurable_space E]
+
+lemma snorm_zero_ae_of_pos {f : α → E} (hp0_lt : 0 < p) (hf : measurable f) (hf_zero : f =ᵐ[μ] 0) :
```

Comment:
I don't think the lemma name really corresponds to the statement. What about `snorm_eq_zero_of_ae_zero` or something like that? And `snorm_eq_zero_of_ae_zero'` for the next one? (The positivity of the exponent, or the nonzeroness of the measure, are not the main conditions of the lemma, so it is not absolutely necessary that they show up in the name).

And for the third one (currently named `snorm_eq_zero`), what about `ae_eq_zero_of_snorm_eq_zero`?

Reply:
I was mimicking other names like `nnorm_zero` (the statement that `nnnorm 0 = 0`) or `ennreal.rpow_zero`. There are many lemma named `fun_zero` to describe the result of the function `fun` applied to 0.

Also `fun_eq_zero` is used in several places for the statement `fun x = 0 -> x = ...`. For example ` pow_eq_zero`. It is also used for the equivalence, for example in `nnnorm_eq_zero` or `abs_eq_zero`.

But I am fine with the names you propose, they fit better with the naming conventions.